x-pack/plugins/reporting/server/routes/internal/management/jobs.ts

```diff
@@ -146,7 +146,7 @@ export function registerJobInfoRoutesInternal(reporting: ReportingCore) {
       {
         path,
         validate: jobHandlers.validate,
-        options: { tags: [ROUTE_TAG_CAN_REDIRECT], access: 'internal' },
```

Comment:
I'm not sure this is the right fix. The bug tells me that our request to download the report does not include the correct header, which is `x-elastic-internal-origin`. The fix should probably go into the front-end code, likely here: 

Looking a bit into the front-end code, we have a button that appears in the "job completion notifier" toast, here: https://github.com/elastic/kibana/blob/9d662b7/x-pack/plugins/reporting/public/notifier/job_download_button.tsx#L23 It looks like it's a plain `href`, so the internal header isn't getting sent in the request. Rather than an `href`, I guess we need to use the core HTTP client, which will automatically provide the internal header.

See: https://github.com/elastic/kibana/pull/162149
> Access to internal Kibana HTTP API endpoints is restricted in serverless and calls to these endpoints will respond with 400 when the request doesn't include an x-elastic-internal-origin header.

> The header is automatically included when using core's browser-side HTTP service (e.g. core.http.fetch('....'))

So, if we use `core.http.fetch()` instead of the plain `href`, that should fix the bug.

Reply:
@tsullivan, we had a couple other places where it wasn't possible to add headers, so we had to make the apis public: 
https://github.com/elastic/kibana/pull/162707
https://github.com/elastic/kibana/pull/162707

> So, if we use core.http.fetch() instead of the plain href, that should fix the bug.

if that endpoint is used to download a file, then you'd have to add a workaround like this: 
https://stackoverflow.com/questions/32545632/how-can-i-download-a-file-using-window-fetch
This is a weird workaround, where you'd first download your file on kibana page in-memory and then you'd download it to a file system. Not sure how large the files can get

Reply:
I chatted with @jloleysens, turns out there is a way now to pass version manually using query param: https://github.com/elastic/kibana/blob/644356c4918b52a420190823f7adba8cecab07e0/packages/core/http/core-http-server/src/versioning/types.ts#L38-L55

I think this is what we need!